### PR TITLE
Rearranging tabs: Unit tests

### PR DESCRIPTION
**Pull request by 0nko**
Task/Issue URL: https://app.asana.com/0/1207418217763355/1207775879598725/f

### Description
This PR adds unit tests to for the new tab manager drag & drop functionality, feature announcement banner and user state detection.

### Steps to test this PR

Running the tests and verifying that everything is green should be enough.

### CODE REVIEW

**Comment by 0nko** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/duckduckgo/Android/4777?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#4777** <a href="https://app.graphite.dev/github/pr/duckduckgo/Android/4777?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#4757** <a href="https://app.graphite.dev/github/pr/duckduckgo/Android/4757?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#4750** <a href="https://app.graphite.dev/github/pr/duckduckgo/Android/4750?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#4743** <a href="https://app.graphite.dev/github/pr/duckduckgo/Android/4743?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#4734** <a href="https://app.graphite.dev/github/pr/duckduckgo/Android/4734?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#4719** <a href="https://app.graphite.dev/github/pr/duckduckgo/Android/4719?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `feature/ondrej/rearranging-tabs`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @0nko and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: app/src/test/java/com/duckduckgo/app/tabs/ui/TabSwitcherViewModelTest.kt [hunk 1]
```diff
@@ -282,13 +292,99 @@ class TabSwitcherViewModelTest {
...
-    fun whenOnDraggingStartedThePixelSent() = runTest {
+    fun whenOnDraggingStartedAnnouncementDismissedAndThePixelSent() = runTest {
         testee.onTabDraggingStarted()
 
         advanceUntilIdle()
 
+        verify(mockTabRepository).setWasAnnouncementDismissed(true)
+
         val params = mapOf("userState" to NEW.name)
         verify(mockPixel).fire(AppPixelName.TAB_MANAGER_REARRANGE_TABS, params, emptyMap(), COUNT)
```
Comment (CrisBarreiro) [inline app/src/test/java/com/duckduckgo/app/tabs/ui/TabSwitcherViewModelTest.kt:303]: Are these the pixels we want to check or is there a specific one for announcement dismissed we want to check as well?

File: app/src/test/java/com/duckduckgo/app/tabs/ui/TabSwitcherViewModelTest.kt [hunk 2]
```diff
@@ -282,13 +292,99 @@ class TabSwitcherViewModelTest {
...
+
+    @Test
+    fun isFeatureAnnouncementVisible_NewUser() = runTest {
+        val isVisible = testee.isFeatureAnnouncementVisible.value
+        assertFalse(isVisible)
+    }
+
+    @Test
+    fun isFeatureAnnouncementVisible_Dismissed() = runTest {
+        `when`(mockTabRepository.tabSwitcherData).thenReturn(flowOf(TabSwitcherData(TabSwitcherData.UserState.EXISTING, true, 0)))
```
Comment (CrisBarreiro) [inline app/src/test/java/com/duckduckgo/app/tabs/ui/TabSwitcherViewModelTest.kt:366]: nit: whenever

File: app/src/test/java/com/duckduckgo/app/userstate/UserStateReporterTest.kt [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+
+    @Test
+    fun verifyUserExistingWhenFirstInstallTimeDoesNotEqualLastInstallTime() = runTest {
+        val packageInfo = PackageInfo().apply {
+            firstInstallTime = 1000L
+            lastUpdateTime = 2000L
+        }
+        initializeSut(packageInfo)
+
+        verify(repository).setIsUserNew(false)
```
Comment (CrisBarreiro) [inline app/src/test/java/com/duckduckgo/app/userstate/UserStateReporterTest.kt:73]: Just curious, did we change the way to check whether a user is new or not? If so, why?

**Review by CrisBarreiro — APPROVED** [review]

File: app/src/test/java/com/duckduckgo/app/userstate/UserStateReporterTest.kt (hunk 3 above)
Comment (0nko) [reply app/src/test/java/com/duckduckgo/app/userstate/UserStateReporterTest.kt:73]: Yes, we did. There was a discussion with @malmstein about this after he posted his feedback in #4743. See my comment here: https://github.com/duckduckgo/Android/pull/4743#issuecomment-2234441154

File: app/src/test/java/com/duckduckgo/app/tabs/ui/TabSwitcherViewModelTest.kt (hunk 1 above)
Comment (0nko) [reply app/src/test/java/com/duckduckgo/app/tabs/ui/TabSwitcherViewModelTest.kt:333]: Yeah, you’re right. Previously there were no banner-specific pixels, but now there are and the drag & drop and dismissing can be tested separated. I’ve refactored this and added a few more tests.

**Pull request merged**